pkg/deploy/generator/resources_rp.go

```diff
@@ -978,25 +978,31 @@ func (g *generator) rpServiceKeyvault() *arm.Resource {
 }
 
 func (g *generator) rpCosmosDB() []*arm.Resource {
-	cosmosdb := &mgmtdocumentdb.DatabaseAccountCreateUpdateParameters{
-		Kind: mgmtdocumentdb.GlobalDocumentDB,
-		DatabaseAccountCreateUpdateProperties: &mgmtdocumentdb.DatabaseAccountCreateUpdateProperties{
-			ConsistencyPolicy: &mgmtdocumentdb.ConsistencyPolicy{
-				DefaultConsistencyLevel: mgmtdocumentdb.Strong,
-			},
-			Locations: &[]mgmtdocumentdb.Location{
+	dbType := sdkcosmos.DatabaseAccountKindGlobalDocumentDB
+	consistency := sdkcosmos.DefaultConsistencyLevelStrong
+	backupPolicy := sdkcosmos.BackupPolicyTypePeriodic
+	minTLSVersion := sdkcosmos.MinimalTLSVersionTls12
```

Comment:
nit: since you are using constants here from the library, I think you don't have to put them in variables explicitly, you can directly put them in the following assignments.

Reply:
Agree. However, the cosmodb object being created were already nested(ugly) so to make it simple to understand I have assigned values to variables and used them instead of directly getting values in the object creation. (Also, to match the way `hashPartitionKey` is being used in other different objects.)

Reply:
If you see on other func definitions like [this](https://github.com/Azure/ARO-RP/blob/1be3f99fdac41a45dc6b0494487f427922b748d6/pkg/deploy/generator/resources_rp.go#L103-L111) , you see that the consts are directly placed while creating/defining an object.
Having variables is a good thing, however in this particular case, someone looking at the object will have to again look up to know the variable initialisation. 
I still think you should put the consts directly rather than through variables.

Reply:
Discussed & agreed upon necessity of variables over call.